internal/cli/server/chains/chain.go

```diff
@@ -19,6 +22,29 @@ var chains = map[string]*Chain{
 }
 
 func GetChain(name string) (*Chain, bool) {
-	chain, ok := chains[name]
-	return chain, ok
+	chain, err := ImportFromFile(name)
+	if err != nil {
+		chain, ok := chains[name]
+		return chain, ok
+	}
```

Comment:
We should reverse the order by first checking `chains[name]` and then from the file because current implementation won't throw any file read/unmarshal error as it will be overridden with the map error.

Reply:
I agree with Manav. This is what polygon-sdk does but I think there are better ways to handle this. For example, right now if there is an error on the chain file, there is no output to the user.

I would check first if the input is a file, then try to unmarshal it like this:

```
func GetChain(name string) (*Chain, error) {
    if _, err := os.Stat(name); err == nil {
      // path exists
      // try to unmarshal
    } else if errors.Is(err, os.ErrNotExist) {
      // path does *not* exist
      // try to read from chains
    } else {
        // filesystem error
       return nil, err
    }
}
```

Reply:
I thought about it but it was written that way in the linear task and I honestly don't think it will make that much of a difference. But the point you have made is correct, In case of incorrect file format it will not give that error to user.

Reply:
Yes, I gave you the link without checking it sorry.

Reply:
Updated